hazardous/metrics/_concordance_index.py

```diff
@@ -0,0 +1,471 @@
+    interest at time :math:`T_i` if:
+
+    - :math:`j` is censored or experience any event at a strictly greater time
+      :math:`T_j > T_i` (pair of type A)
+    - :math:`j` is censored at the exact same time :math:`T_i = T_j` (pair of type A).
+    - :math:`j` experiences a competing event before or at time :math:`T_i`
+      (pair of type B)
+
+    A pair is then considered concordant if the predicted incidence of the event of
+    interest for :math:`i` at time :math:`\tau` is larger than the predicted incidence
+    for :math:`j`. If both predicted incidences are ties, the pair is counted as
+    :math:`1/2` for the count of concordant pairs.
+
+    The C-index has been shown to depend on the censoring distribution, and an
+    inverse probability of censoring weighting (IPCW) allows to overcome this
+    limitation [Uno2011]_, [Gerds2013]_. By default, the IPCW is implemented with a
+    Kaplan-Meier estimator. Additionnally, the C-index is not a proper metric to
+    evaluate a survival model [Blanche2019]_, and alternatives as the integrated Brier
+    score (``integrated_brier_score_incidence``) should be considered.
+    The C-index is a ranking metric and, unlike proper scoring rule metrics,
+    cannot evaluate the calibration of the estimator.
+
+    Parameters
+    ----------
+    y_test : array, dictionnary or dataframe of shape (n_samples, 2)
+        The test target, consisting in the 'event' and 'duration' columns.
+
+    y_pred: array of shape (n_samples, n_time_grid)
+        Cumulative incidence for the event of interest, at the time points
+        from the input time_grid.
+
+    y_train : array, dictionnary or dataframe of shape (n_samples, 2), default=None
+        The train target, consisting in the 'event' and 'duration' columns.
+        Only used when ipcw_estimator = "km".
+
+    ipcw_estimator : None, 'km', or fitted estimator, default="km"
+        The inverse probability of censoring weighted (IPCW) estimator.
+
+        - Pass None to set uniform weights to all samples.
+        - Pass "km" to use the Kaplan-Meier IPCW estimator. It fits using y_train,
+          which must be set.
+
+    time_grid: array of shape (n_time_grid,), default=None
+        Time points used to predict the cumulative incidence.
+
+    taus: float or array of shape (n_taus,), default=None
+        Timepoint(s) at which the concordance index is evaluated.
+
+    event_of_interest: int, default=1
+        For competing risks, the event of interest.
+
+    tied_tol : float, default=1e-8
+        The tolerance range to consider two probabilities equal.
+
+    Returns
+    -------
+    cindex: array of shape (n_taus,)
+        Value of the concordance index for each tau in taus.
+
+    References
+    ----------
+    .. [Wolbers2014] M. Wolbers, P. Blanche, M. T. Koller, J. C. Witteman, T. A. Gerds,
+       "Concordance for prognostic models with competing risks", 2014
+
```

Comment:
those links are clickable, but refer to the same page, maybe they should link to the article?